### PR TITLE
Watchlist Filters PR Feedback

### PR DESCRIPTION
**Pull request by tonisevener**
**Phabricator:** 
https://phabricator.wikimedia.org/T342929#9148645
https://phabricator.wikimedia.org/T342929#9150414

### Notes
This PR addresses design feedback and a bug:

1. Done button has been changed to match Notification Filters Done button weight. I also changed it to the theme link color.
2. I noticed theming was buggy, particularly if the system theme did not match the theme chosen in app Settings. 
3. Removed the "All" option from the Latest Revisions section.

~~Dependent PR is https://github.com/wikimedia/wikipedia-ios-data/pull/11~~ merged.

### CODE REVIEW

File: Sources/Components/Components/Project Icons/WKProjectIconView.swift [hunk 1]
```diff
@@ -28,7 +28,7 @@ public struct WKProjectIconView: View {
             )
 
         case .wikidata:
-            if let image = UIImage(named: "project-commons", in: Bundle.module, compatibleWith: nil) {
+            if let image = UIImage(named: "project-wikidata", in: Bundle.module, compatibleWith: nil) {
```
Comment (staykids) [inline Sources/Components/Components/Project Icons/WKProjectIconView.swift:31]: 😅 Nice catch on these two images

File: Sources/Components/Components/Watchlist/Filter/WKWatchlistFilterView.swift [hunk 2]
```diff
@@ -22,7 +22,8 @@ struct WKWatchlistFilterView: View {
                             doneAction()
                         }) {
                             Text(viewModel.localizedStrings.doneTitle)
-                                .foregroundColor(Color(theme.text))
+                                .fontWeight(Font.Weight.semibold)
```
Comment (staykids) [inline Sources/Components/Components/Watchlist/Filter/WKWatchlistFilterView.swift]: This should probably flow through `WKFont`.

**Review by staykids — COMMENTED** [review]
Nice - minor font note below.

File: Sources/Components/Components/Watchlist/Filter/WKWatchlistFilterView.swift (hunk 2 above)
Comment (tonisevener) [reply Sources/Components/Components/Watchlist/Filter/WKWatchlistFilterView.swift]: Agh, good catch. I pushed a fix.

**Review by staykids — APPROVED** [review]

**Pull request merged**